lib/src/Uncertainty/Algorithm/Simulation/CrossEntropyImportanceSampling.cxx

```diff
@@ -0,0 +1,245 @@
+  Sample auxiliaryInputSample = auxiliaryDistribution_.getSample(numberOfSample);
+
+  // Evaluation on limit state function
+  Sample auxiliaryOutputSample = computeOutputSamples(auxiliaryInputSample);
+  
+  // Computation of current quantile
+  Scalar currentQuantile = auxiliaryOutputSample.computeQuantile(rhoQuantile_)[0];
+  
+  Point  auxiliaryDistributionParameters;
+
+  if (getEvent().getOperator()(currentQuantile, getEvent().getThreshold()))
+  {
+    currentQuantile = getEvent().getThreshold();
+    //auxiliaryDistribution_ = initialDistribution_;
+  }
+  else
+  {
+  Indices indiceCritic(0);
+
+    for (UnsignedInteger i = 0; i < auxiliaryInputSample.getSize(); ++i)
+      {
+        const Bool weightBool = getEvent().getOperator()(auxiliaryOutputSample(i, 0), currentQuantile);
+        if (weightBool)
+          indiceCritic.add(i);
+      } 
+
+     const Sample auxiliaryCriticInputSamples(auxiliaryInputSample.select(indiceCritic));
+     
+   // Optimize auxiliary distribution parameters
+    auxiliaryDistributionParameters = optimizeAuxiliaryDistributionParameters(auxiliaryCriticInputSamples);
+
+    // Update auxiliary Distribution Parameters
+    updateAuxiliaryDistribution(auxiliaryDistributionParameters); 
+
+  }
+
+  UnsignedInteger iterationNumber  = 0;
+  
+  while ((getEvent().getOperator()(getEvent().getThreshold(), currentQuantile)) && (currentQuantile != getEvent().getThreshold()))
+  {
+    ++iterationNumber ;
+    
+    Point currentAuxiliaryDistributionParameters = auxiliaryDistributionParameters;
+
+    // Drawing of samples using auxiliary density and evaluation on limit state function
+    auxiliaryInputSample = Sample(0, initialDistribution_.getDimension());
+    auxiliaryOutputSample = Sample(0, 1);
+
+
+
+    for (UnsignedInteger i = 0; i < getMaximumOuterSampling(); ++i)
+    {
+      const Sample blockSample(auxiliaryDistribution_.getSample(getBlockSize()));
+      auxiliaryInputSample.add(blockSample);
+      auxiliaryOutputSample.add(computeOutputSamples(blockSample));
+
+      if (stopCallback_.first && stopCallback_.first(stopCallback_.second))
+        throw InternalException(HERE) << "User stopped simulation";
+    }
+
+    // Computation of current quantile
+    currentQuantile = auxiliaryOutputSample.computeQuantile(rhoQuantile_)[0];
+    
+    // If failure probability reached, stop the adaptation
+    if (getEvent().getOperator()(currentQuantile, getEvent().getThreshold()))
+    {
+      currentQuantile = getEvent().getThreshold();
+    }
+    else
+    {
+    Indices indiceCritic(0);
+    for (UnsignedInteger i = 0; i < auxiliaryInputSample.getSize(); ++i)
+      {
+        const Bool weightBool = getEvent().getOperator()(auxiliaryOutputSample(i, 0), currentQuantile);
+        if (weightBool)
+          indiceCritic.add(i);
+      } 
+
+   // Extract the relevant sample
+   const Sample auxiliaryCriticInputSamples(auxiliaryInputSample.select(indiceCritic));
+  
```

Comment:
Please removed unnecessary blank lines. I think that classical OTish code does not have any blank line, at all.
```suggestion
```